src/archivematicaCommon/lib/archivesspace/client.py

```diff
@@ -0,0 +1,424 @@
+import json
+import os
+import requests
+from urlparse import urlparse
+
+
+class ConnectionError(Exception):
+    pass
+
+
+class AuthenticationError(Exception):
+    pass
+
+
+class CommunicationError(Exception):
+    def __init__(self, status_code, response):
+        message = "ArchivesSpace server responded {}".format(status_code)
+        self.response = response
+        super(CommunicationError, self).__init__(message)
+
+
+class ArchivesSpaceError(Exception):
+    pass
+
+
+class ArchivesSpaceClient(object):
+    """
+    Client to communicate with a remote ArchivesSpace installation using its backend API.
+
+    Note that, while functions follow the same API as the ArchivistsToolkitClient, one major difference is the handling of resource and component IDs.
+    In ArchivistsToolkitClient, resource IDs are longs representing the database row ID.
+    In this client, resource IDs are instead URI fragments representing the location of the record, for instance:
+        /repositories/2/resource/1
+    This change is due to the fact that the integer IDs are not unique across the collection in ArchivesSpace, as they were in Archivist's Toolkit.
+    """
+
+    RESOURCE = 'resource'
+    RESOURCE_COMPONENT = 'resource_component'
+
+    def __init__(self, host, user, passwd, port=8089, repository=2):
+        parsed = urlparse(host)
+        if not parsed.scheme:
+            host = 'http://' + host
+
+        self.host = host + ':' + str(port)
+        self.user = user
+        self.passwd = passwd
+        self.repository = '/repositories/{}'.format(repository)
+        self._login()
+
+    def _login(self):
+        try:
+            response = requests.post(self.host + '/users/' + self.user + '/login',
+                                     data={'password': self.passwd})
+        except requests.ConnectionError as e:
+            raise ConnectionError("Unable to connect to ArchivesSpace server: " + str(e))
+
+        try:
+            output = response.json()
+        except:
```

Comment:
Looks like the surface area where this could come up is limited, but it's possible for the JSON decoder to not be simplejson (and hence for the exception to be a different class). Think I'll leave this alone.